File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsJava/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicsjava/data/adapter/UserViewHolder.java
package pl.udu.uwr.pum.offlinecachingbasicsjava.data.adapter;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import pl.udu.uwr.pum.offlinecachingbasicsjava.data.User;
import pl.udu.uwr.pum.offlinecachingbasicsjava.databinding.RvItemBinding;

public class UserViewHolder extends RecyclerView.ViewHolder {

    private final RvItemBinding binding;

    public UserViewHolder(RvItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(User item){
        binding.firstName.setText(item.getFirst_name());
        binding.lastName.setText(item.getLast_name());
        binding.city.setText(item.getAddress().getCity());
        binding.country.setText(item.getAddress().getCountry());
        binding.dateOfBirth.setText(item.getDate_of_birth());
        binding.creditCardNumber.setText(item.getCredit_card().getCc_number());
        binding.email.setText(item.getEmail());
        binding.employment.setText(item.getEmployment().getTitle());
        binding.gender.setText(item.getGender());
        binding.password.setText(item.getPassword());
        binding.paymentMethod.setText(item.getSubscription().getPayment_method());
        binding.phone.setText(item.getPhone_number());
        binding.plan.setText(item.getSubscription().getPlan());
        binding.term.setText(item.getSubscription().getTerm());
        binding.paymentMethod.setText(item.getSubscription().getPayment_method());
        binding.state.setText(item.getAddress().getState());
        binding.zipCode.setText(item.getAddress().getZip_code());
        binding.status.setText(item.getSubscription().getStatus());
        binding.streetAddress.setText(item.getAddress().getStreet_address());
        binding.streetName.setText(item.getAddress().getStreet_name());
        binding.username.setText(item.getUsername());
        Glide.with(binding.getRoot()).load(item.getAvatar()).into(binding.image);
    }
}

File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicskotlin/data/adapters/UserViewHolder.kt
package pl.udu.uwr.pum.offlinecachingbasicskotlin.data.adapters

import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import pl.udu.uwr.pum.offlinecachingbasicskotlin.data.User
import pl.udu.uwr.pum.offlinecachingbasicskotlin.databinding.RvItemBinding

class UserViewHolder(private val binding: RvItemBinding) : RecyclerView.ViewHolder(binding.root) {
    fun bind(item: User){
        binding.firstName.text = item.first_name
        binding.lastName.text = item.last_name
        binding.city.text = item.address.city
        binding.country.text = item.address.country
        binding.dateOfBirth.text = item.date_of_birth
        binding.creditCardNumber.text = item.credit_card.cc_number
        binding.email.text = item.email
        binding.employment.text = item.employment.title
        binding.gender.text = item.gender
        binding.password.text = item.password
        binding.paymentMethod.text = item.subscription.payment_method
        binding.phone.text = item.phone_number
        binding.plan.text = item.subscription.plan
        binding.term.text = item.subscription.term
        binding.paymentMethod.text = item.subscription.payment_method
        binding.state.text = item.address.state
        binding.zipCode.text = item.address.zip_code
        binding.status.text = item.subscription.status
        binding.streetAddress.text = item.address.street_address
        binding.streetName.text = item.address.street_name
        binding.username.text = item.username
        Glide.with(binding.root).load(item.avatar).into(binding.image)
    }
}